intern/convenience/array.py

```diff
@@ -0,0 +1,369 @@
+"""
+Copyright 2018-2020 The Johns Hopkins University Applied Physics Laboratory.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+# Standard imports
+from typing import Union, Tuple
+from collections import namedtuple
+
+# Pip-installable imports
+import numpy as np
+
+import blosc
+import requests
+
+from intern.resource.boss.resource import (
+    ChannelResource,
+    CoordinateFrameResource,
+    ExperimentResource,
+)
+from intern.remote.boss import BossRemote
+from intern.utils.parallel import block_compute
+
+# A named tuple that represents a bossDB URI.
+bossdbURI = namedtuple(
+    "bossdbURI", ["collection", "experiment", "channel", "resolution"]
+)
+
+
+class VolumeProvider:
+    def get_channel(self, channel: str, collection: str, experiment: str):
```

Comment:
These type annotations make the rest of intern look naked! 😀

Reply:
heh, type-anno'ing the rest of the library is gonna be a long journey...